pkg/agent/informant.go

```diff
@@ -1091,3 +1175,93 @@ func (s *InformantServer) Upscale(ctx context.Context, logger *zap.Logger, to ap
 	logger.Info("Received successful upscale result")
 	return nil
 }
+
+func (s *InformantServer) MonitorHealthCheck(ctx context.Context, logger *zap.Logger) error {
```

Comment:
this function (and `MonitorUpscale` / `MonitorDownscale` should have comments explaining their usage).

Also: consider adding requestLock usage here as well, plus facilities to restart the connection (via `s.exit`) if the request fails.

Reply:
What would the advantages of using requestLock here be? All the types involved (`Dispatcher` and `CondChannelSignal`) are thread-safe.

Reply:
The general idea behind having requestLock was to avoid making concurrent requests to the informant (because e.g. concurrent "upscale" and "downscale" could have unpredictable consequences). In practice, I think we _should_ be fine without it, but we lose some nice guarantees - in particular, if we e.g. call `server.exit(...)` while holding requestLock, we know that no further requests will be made.

This was more relevant for the informant, where the /register request was handled more like other requests, so we _did_ actually have to prevent concurrent register attempts. But it may be useful to keep here as well.